src/pip/_internal/network/session.py

```diff
@@ -362,12 +363,24 @@ def __init__(
             backoff_factor=0.25,
         )  # type: ignore
 
+        # Used to set numbers of parallel downloads in
+        # pip._internal.network.BatchDownloader and to set pool_connection in
+        # the HTTPAdapter to prevent connection pool from hitting the default(10)
+        # limit and throwing 'Connection pool is full' warnings
+        self.parallel_downloads = (
+            parallel_downloads if (parallel_downloads is not None) else 1
+        )
+        pool_maxsize = max(self.parallel_downloads, 10)
```

Comment:
Is this not a configurable limit? It seems reasonable to allow more than 10 threads for downloading as each package should be able to be downloading in parallel.

Reply:
Edit: I see now that we actually do set it below, so I think this is just unnecessary now.

Please consider removing changing to just
`pool_maxsize = self.parallel_downloads`

Reply:
The default pool_maxsize is 10 in requests.Session. My thought process was to not change it if it wasn't necessitated by the user asking for more than 10 connections to be opened

Reply:
Thanks Neil, you're right. On a second review that seems a wise decision. I actually misread it as `min` not `max`. Finally got some coffee and this looks ready to land if the project maintainers can review it.